stowsh

```diff
@@ -50,6 +50,35 @@ stowsh_setpaths() {
         deperr "GNU findutils"
         return 1
     fi
+    if [[ "${USEGIT}" -eq 1 ]]; then
+        flist="list_git_files"
+    else
+        flist="list_files"
+    fi
+}
+
+list_files() {
+    if [[ "$1" == "-d" ]];
+    then
+        $findcmd . -mindepth 1 -type d -printf "%P\0"
+    else
+        $findcmd . -type f -printf "%P\0" -or -type l -printf "%P\0"
+    fi
+}
+
+list_git_files() {
+    gitfiles=( `git ls-files` )
+    if [[ "${#gitfiles[@]}" == 0 ]]; then return; fi
+
+    if [[ "$1" == "-d" ]];
+    then
+        dirname "${gitfiles[@]}" |\
```

Comment:
Unfortunately this doesn't work on macOS, where the BSD version of dirname doesn't accept more than one argument. If possible, I would prefer this to be rewritten as a loop or with xargs to accumulate the result of multiple calls. This would avoid adding a coreutils dependency (see #14).

But, if you can't figure it out ... stowsh already has GNU coreutils as a requirement, and relying on GNU behavior here wouldn't make the situation significantly worse than it already is. If you stick with GNU dirname then please follow the pattern in `stowsh_setpaths` and `isgnu` to set it up.

The general idea is that the script should first check for `g<utility>` and use that if it exists (because homebrew installs GNU coreutils with a `g` prefix by default). And then it should check `<utility>` is indeed the GNU version (and abort if not).

Reply:
If you go with `xargs` bear in mind that the BSD xargs on macOS doesn't support long GNU switches, but I think it does what you need, i.e.
```
$ paths=(foo/1 bar/2 baz/3)
$ printf "%s\n" ${paths[@]} | xargs -n1 dirname
foo
bar
baz
```

Reply:
I will go the xargs way, just let me find 5 minutes to test and commit.